go/vt/tabletserver/sqlquery.go

```diff
@@ -175,12 +190,12 @@ func (sq *SqlQuery) disallowQueries() {
 	sq.mu.Unlock()
 	// Terminate all streaming queries
 	sq.qe.streamQList.TerminateAll()
-	// Don't hold lock while waiting.
+	// Wait for outstanding requests to finish.
 	sq.requests.Wait()
 
-	// SHUTTING_QUERIES -> NOT_SERVING
-	sq.mu.Lock()
 	defer func() {
```

Comment:
It should recover gracefully on panic because all public entry points are protected: https://github.com/youtube/vitess/blob/master/go/vt/tabletserver/queryctl.go#L294.
This is why I do log.Fatal to really force an exit. We have some higher level issues to discuss:
1. We currently use panic as error mechanism. I think we should remove this.
2. Serendipitously, designing tabletserver like this has made it resilient to unexpected panics. We have to discuss if we should continue to do this even if we get rid of the panic mechanism.

Calling on @henryanand for the second pair of eyeballs.